keyboards/sofle/keymaps/bastien-baloup_fr/keymap.c

```diff
@@ -0,0 +1,218 @@
+ * | Tab  |   q  |   s  |   d  |   f  |  g   |-------.     ,-------| PgDo | Left | Down | Rigth|CALeft|CARigt|
+ * |------+------+------+------+------+------|       |     | Play  |------+------+------+------+------+------|
+ * |Shift |   w  |   x  |   c  |   v  |  b   |-------|     |-------|      | PWrd |      | NWrd | Menu | Shift|
+ * `-----------------------------------------/      /       \      \-----------------------------------------'
+ *            | LGUI | LAlt | LCTR |  MO2 | /Enter  /        \Space \  |  MO3 | RCTR | RAlt | HYPER|
+ *            |      |      |      |      |/      /           \      \ |      |      |      |      |
+ *            `----------------------------------'             '------''---------------------------'
+ */
+[3] = LAYOUT(
+  _______, _______, _______, _______, _______, _______,                         KC_CAPS, KC_PSCR   , KC_SCRL, KC_PAUS   , KC_INS       , _______      ,
+  _______, _______, _______, _______, _______, _______,                         KC_PGUP, KC_HOME   , KC_UP  , KC_END    , KC_DEL       , _______      ,
+  _______, _______, _______, _______, _______, _______,                         KC_PGDN, KC_LEFT   , KC_DOWN, KC_RGHT   , C(A(KC_LEFT)), C(A(KC_RGHT)),
+  KC_LSFT, _______, _______, _______, _______, _______, _______,       KC_MPLY, XXXXXXX, C(KC_LEFT), XXXXXXX, C(KC_RGHT), KC_MENU      , KC_RSFT      ,
+                  _______, _______, _______, _______, _______,           _______, _______, _______, _______, _______
+)
+};
+
+// Custom keycode handling.
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+    // handling this once instead of in each keycode uses less program memory.
+    if ((keycode >= SAFE_RANGE) && !(record->event.pressed)) {
+        return false;
+    }
+
+    switch (keycode) {
+        case CL_AGRV:
+            // capsLockOn à capsLockOff
+            SEND_STRING(SS_TAP(X_CAPS)"0"SS_TAP(X_CAPS));
+            break;
+        case CL_EACU:
+            // capsLockOn é capsLockOff
+            SEND_STRING(SS_TAP(X_CAPS)"2"SS_TAP(X_CAPS));
+            break;
+        case CL_EGRV:
+            // capsLockOn è capsLockOff
+            SEND_STRING(SS_TAP(X_CAPS)"7"SS_TAP(X_CAPS));
+            break;
+        case UC_PI:
+            // Ctrl+Shift+u 3CO Enter
+            tap_code16(LCTL(LSFT(KC_U)));
+            tap_code16(S(KC_3));
+            tap_code16(KC_C);
+            tap_code16(S(KC_0));
+            tap_code16(KC_ENT);
+            break;
+        case UC_ELIP:
+            // Ctrl+Shift+u 2026 Enter
+            tap_code16(LCTL(LSFT(KC_U)));
+            tap_code16(S(KC_2));
+            tap_code16(S(KC_0));
+            tap_code16(S(KC_2));
+            tap_code16(S(KC_6));
+            tap_code16(KC_ENT);
+            break;
+    }
+
+    // this uses less memory than returning in each case.
+    return keycode < SAFE_RANGE;
+};
+
+#ifdef ENCODER_ENABLE
+#ifdef ENCODER_MAP_ENABLE
```

Comment:
Should only need to check encoder map, as it shouldn't get defined unless encoder is also enabled:
```suggestion
#ifdef ENCODER_MAP_ENABLE
```

worst case:
```suggestion
#if defined(ENCODER_ENABLE) && defined(ENCODER_MAP_ENABLE)
```